Sources/GraphQLWebSocket/WebSocketClient.swift

```diff
@@ -0,0 +1,200 @@
+//
+//  WebSocketClient.swift
+//
+//
+//  Created by Michał A on 2023/10/25.
+//
+
+// MARK: - Starscream types
+
+// see also https://github.com/daltoniam/Starscream/blob/master/Sources/Starscream/WebSocket.swift
+
+import Foundation
+
+public protocol WebSocketClient: AnyObject {
+    func connect()
+    func disconnect(closeCode: UInt16)
+    func write(string: String, completion: (() -> ())?)
+    func write(stringData: Data, completion: (() -> ())?)
+    func write(data: Data, completion: (() -> ())?)
+    func write(ping: Data, completion: (() -> ())?)
+    func write(pong: Data, completion: (() -> ())?)
+}
+
+extension WebSocketClient {
+    func write(string: String) {
+        write(string: string, completion: nil)
+    }
+    
+    func write(data: Data) {
+        write(data: data, completion: nil)
+    }
+    
+    func write(ping: Data) {
+        write(ping: ping, completion: nil)
+    }
+    
+    func write(pong: Data) {
+        write(pong: pong, completion: nil)
+    }
+    
+    func disconnect() {
+        disconnect(closeCode: UInt16(URLSessionWebSocketTask.CloseCode.normalClosure.rawValue))
+    }
+    
+    func write(stringData: Data, completion: (() -> ())?) {
+        let string = String(data: stringData, encoding: .utf8)!
+        write(string: string, completion: completion)
+    }
+    
+    func write(stringData: Data) {
+        write(stringData: stringData, completion: nil)
+    }
+}
+
+public enum WebSocketEvent {
+    case connected([String: String])
+    case disconnected(String, UInt16)
+    case text(String)
+    case binary(Data)
+    case pong(Data?)
+    case ping(Data?) // NOTE: not used
+    case error(Error?)
+    case viabilityChanged(Bool) // NOTE: not used
+    case reconnectSuggested(Bool) // NOTE: not used
+    case cancelled // NOTE: not used
+    case peerClosed // NOTE: not used
```

Comment:
I'm not a fan of including unused code. Is this simply incomplete? I assume not since you're opening a PR. Can we simple remove them?

Reply:
This was defined by Starscream and as of it is "used" by `GraphQLWebSocket` even though it only prints debug messages for some of them:

```swift            
        case .viabilityChanged(_):
            self.config.logger.debug("Server viability changed...")
            break
            
        case .reconnectSuggested(_):
            self.config.logger.debug("Server suggested reconnect...")
            break
```

I plan to update the Starscream types in next iteration, but first I wanted to keep `GraphQLWebSocket` intact and reuse Starscream interfaces.